image/image.go

```diff
@@ -29,3 +35,27 @@ type Image struct {
 	URL        string `json:"url,omitempty"`
 	Base64JSON string `json:"b64_json,omitempty"`
 }
+
+func (c *Image) Save(filename string, toBase64 bool) (ok bool, err error) {
```

Comment:
Base64JSON is populated by the server when response type is set to `b64_json`. We shouldn't mutate it.

On the other hand, instead of assuming that the user have access to a file system, we should turn this method into something that returns an io.ReadCloser for flexibility. What about something like below?

```
func (i *Image) Reader() (io.ReadCloser, error) {
	if i.URL != "" {
		resp, err := http.Get(i.URL)
		if err != nil {
			return nil, err
		}
		return resp.Body, nil
	}
	if i.Base64JSON != "" {
		decoded, err := base64.RawStdEncoding.DecodeString(i.Base64JSON)
		if err != nil {
			return nil, err
		}
		return ioutil.NopCloser(bytes.NewBuffer(decoded)), nil
	}
	return nil, errors.New("no image data")
}
```

Reply:
We should use `base64.StdEncoding` since `RawStdEncoding` use `nopadding` and would cause panic.